## Inferring Chromosome Radial Organization from Hi-C Data ##

The included Python scripts (mat2net.py, net2dist.py, and dist2rad.py) contain an implementation of the computational analysis technique described in the following manuscript.

Manuscript

The computational technique first generates multiple 3D network configurations from bulk Hi-C data using a force-directed graph layout algorithm and then incorporates gene density and chromosome length properties to infer radial territorial position of chromosomes inside the nucleus.

When using the Hi-C contact data as an input, the algorithm expects a square contact matrix. 

The analysis technique can be executed by running the following commands in three steps - 

python mat2net.py -i GM12878-Control-R2__hg19__genome__C-2500000-iced.matrix.gz -n 100 -o test_dir

-i : input file name in .matrix.gz format
-n : number of samples (network structures)
-o : location of the output directory for generated samples

python net2dist.py -l GM12878-Control-R2__hg19__genome__C-2500000-iced.matrix.gz.label -n 100 -g GM12878-Control-R2__hg19__genome__C-2500000-iced.matrix.gz.gexf -s test_dir 

-l : input file name in .label format (generated using the first command)
-n : number of samples (network structures)
-g : input network configuration file name in .gexf format (generated using the first command)
-s : location of samples as input

python dist2rad.py -W ignore -i GM12878-Control-R2__hg19__genome__C-2500000-iced.matrix.gz -d test_dirDist.txt 

-i : input file name in .matrix.gz format
-d : input network model distances in .txt format (generated using the second command)

A sample dataset (GM12878-Control-R2__hg19__genome__C-2500000-iced.matrix.gz) has been provided, which is a processed version of GM12878 dataset generated by Sanders et al. [1].

### Requirements ###

* NumPy (1.19.0)
* SciPy (1.5.0)
* Pandas (1.0.3)
* NetworkX (2.3)
* Scikit-learn (0.22.2)
* Loess (2.0.11)
* Kneed (0.6.0)

[1] Sanders, Jacob T., et al. "Radiation-Induced DNA Damage and Repair Effects on 3D Genome Organization." bioRxiv (2019): 740704.

